### ISSUE TITLE
Only update portion of azure_rm_appgateway configuration?

### ISSUE THREAD

**Issue by tfmark** [ansible/ansible_collections_azure#992]
##### SUMMARY
I would like to be able to update the SSL certificate of my Application Gateway

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
azure.azcollection.azure_rm_appgateway

##### ADDITIONAL INFORMATION

From my unsuccessful testing this afternoon I believe I need to submit a _full_ configuration item to azure_rm_appgateway rather than being able to simply update an individual piece of configuration of an existing gateway (in this case I want to setup a new SSL certificate.

From what I can see, the call to `create_update_applicationgateway` does not take into consideration the `oldresponse` (and does not do a delta) which means I need to include the entire configuration for my app gateway (I guess there is an argument that I should have created the gateway via ansible in the first place..)

**Comment by Fred-sun** [comment]
@tfmark You are welcome to submit the problems encountered, we will investigate and deal with the problems as soon as possible! Thank you very much!

**Comment by akira6592** [comment]
I too would be happy to be able to partial update with the `azure_rm_appgateway` module.

Currently, I think partial updates can only be done through the [Azure CLI](https://learn.microsoft.com/ja-jp/cli/azure/network/application-gateway?view=azure-cli-latest).
Since the API cannot PATCH the settings, the Azure CLI implementation seems to be to GET all settings once and PUT them back.

**Comment by emanueleaina** [comment]
Indeed. What makes this particularly painful is that at the moment doing a roundtrip with `azure_rm_appgateway_info` is impossible since it only returns a very very basic subset of the `azure_rm_appgateway` configuration.

My use case is that I want to use Application Gateway as Ingress Controller (AGIC) from AKS, but I want to create the AppGw myself to enable WAF. I absolutely do not want to poke backends and probes from Ansible since that's what AGIC does, but I still want to be able to tweak the other settings.

However I am effectively unable to do so:

* I can't provision a new gateway with `azure_rm_appgateway` with no `frontend_ports` and other settings as they are required (by the API)
  * I put some placeholder that AGIC will delete, this is fine
* I can't update a subset of the settings with `azure_rm_appgateway` (for instance, `web_application_firewall_configuration`) for the same reason
 * this is **not** fine, using placeholders would needlessly disrupt the gateway until AGIC syncs it again
* `azure_rm_appgateway_info` does not round-trip with  `azure_rm_appgateway` so I cannot implement partial updates with that
* getting the json from `az network application-gateway show` leads to death by papercuts
  * some values differ from what  `azure_rm_appgateway` enforces:
    * `value of protocol must be one of: http, https, got: Http found in http_listeners`
    * `value of protocol must be one of: http, https, got: Http found in backend_http_settings_collection`
  * the json has more fields than what `azure_rm_appgateway` expects:
   * invalid backendAddressPool fields: backendAddresses, etag, id, provisioningState, resourceGroup, type
   * invalid frontendPort fields: etag, id, provisioningState, resourceGroup, type
   * invalid httpListener fields: etag, frontendIPConfiguration, frontendPort, hostNames, id, provisioningState, requireServerName fndication, resourceGroup, sslCertificate, type,
   * invalid RequestRoutingRule fields: etag, httpListener, id, provisioningState, resourceGroup, ruleType, type, urlPathMap
   * invalid backendHttpSettingsCollection fields: affinityCookieName, cookieBasedAffinity, etag, id, pickHostNameFromBackendAddress, provisioningState, requestTimeout, resourceGroup, type
 * the `probe: {id: "/…/…" }` field is serialized as a string as it assumes it is just the probe name, leading to resource ids like `/subscriptions/…/resourceGroups/…/providers/Microsoft.Network/applicationGateways/…/probes/{"id": "/subscriptions/…/resourceGroups/…/providers/Microsoft.Network/applicationGateways/…/probes/myProbe"}`
 * even fixing all of that leads to `Resource /subscriptions/…/resourceGroups/…/providers/Microsoft.Network/applicationGateways/…/probes/myProbe referenced by resource /subscriptions/…/resourceGroups/…/providers/Microsoft.Network/applicationGateways/…/backendHttpSettingsCollection/… was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.` while the probe id is the one returned by `az network application-gateway show`

At the moment I got stuck on the last point. I am now planning to discard AGIC and potentially Application Gateway completely and just switch to a different ingress controller with its own WAF.
